wrappers/python/xraydb/xraydb.py

```diff
@@ -205,40 +222,46 @@ def query(self, *args, **kws):
         return self.session.query(*args, **kws)
 
     def f0_ions(self, element=None):
-        """return list of ion names supported for the .f0() calculation
+        """
+        return list of ion names supported for the .f0() calculation
         from Waasmaier and Kirfel
 
-        arguments
-        ---------
-        element:  atomic number, atomic symbol or ionic symbol
-                  (case insensitive) of scatterer
+        Parameters:
+            element (string, int, pr None):  atomic number, symbol, or ionic symbol
+                    of scattering element.
+
+        Returns:
+            if element is None, all 211 ions are returned.
+            if element is not None, the ions for that element are returned
 
-        if element is None, all 211 ions are returned.  If element is
-        not None, the ions for that element (atomic symbol) are returned
+        Example:
+             >>> xdb = XrayDB()
+             >>> xdb.f0_ions('Fe')
+             ['Fe', 'Fe2+', 'Fe3+']
+
+        Notes:
+           Z values from 1 to 98 (and symbols 'H' to 'Cf') are supported.
         """
         rows = self.query(WaasmaierTable)
         if element is not None:
-            if isinstance(element, int):
-                rows = rows.filter(WaasmaierTable.atomic_number==element)
-            else:
-                rows = rows.filter(WaasmaierTable.element==element.title())
+            rows = rows.filter(WaasmaierTable.element==self.symbol(element))
         return [str(r.ion) for r in rows.all()]
 
     def f0(self, ion, q):
-        """Calculate f0(q) -- elastic x-ray scattering factor
-        from Waasmaier and Kirfel
+        """
+        return f0(q) -- elastic X-ray scattering factor from Waasmaier and Kirfel
 
-        arguments
-        ---------
-        ion:  atomic number, atomic symbol or ionic symbol
-              (case insensitive) of scatterer
+        Parameters:
+            ion (string, int, or None):  atomic number, symbol or ionic symbol
+                  of scattering element.
+            q (float, list, ndarray): value(s) of q for scattering factors
 
-        q: single q value, list, tuple, or numpy array of q value
-             q = sin(theta) / lambda
-             theta = incident angle, lambda = x-ray wavelength
+        Returns:
+            ndarray of elastic scattering factors
 
-        Z values from 1 to 98 (and symbols 'H' to 'Cf') are supported.
-        The list of ionic symbols can be read with the function .f0_ions()
+        Notes:
+            q = sin(theta) / lambda, where theta = incident angle, and
```

Comment:
Hm, might be more complicated, and need more documentation.  "sin(theta)/lambda" is what used in the original source, but is called "k".  I believe it might actually be that scattering "q" may actually be "sin(theta)/lambda".... This may need advice from an Acta Cryst A level scatterer.  Will investigate.